src/Util/Message.cc

```diff
@@ -0,0 +1,340 @@
+    spatial_config.set_start(start);
+    spatial_config.set_end(end);
+    spatial_config.set_mip(mip);
+    return spatial_config;
+}
+
+CARTA::IntBounds Message::IntBounds(int min, int max) {
+    CARTA::IntBounds int_bounds;
+    int_bounds.set_min(min);
+    int_bounds.set_max(max);
+    return int_bounds;
+}
+
+CARTA::FloatBounds Message::FloatBounds(float min, float max) {
+    CARTA::FloatBounds float_bounds;
+    float_bounds.set_min(min);
+    float_bounds.set_max(max);
+    return float_bounds;
+}
+
+CARTA::MomentRequest Message::MomentsRequest(int32_t file_id, int32_t region_id, CARTA::MomentAxis moments_axis,
+    CARTA::MomentMask moment_mask, CARTA::IntBounds spectral_range, CARTA::FloatBounds pixel_range) {
+    CARTA::MomentRequest moment_request;
+    moment_request.set_file_id(file_id);
+    moment_request.set_region_id(region_id);
+    moment_request.set_axis(moments_axis);
+    moment_request.set_mask(moment_mask);
+    auto* mutable_spectral_range = moment_request.mutable_spectral_range();
+    mutable_spectral_range->set_min(spectral_range.min());
+    mutable_spectral_range->set_max(spectral_range.max());
+    auto* mutable_pixel_range = moment_request.mutable_pixel_range();
+    mutable_pixel_range->set_min(pixel_range.min());
+    mutable_pixel_range->set_max(pixel_range.max());
+    moment_request.add_moments(CARTA::Moment::MEAN_OF_THE_SPECTRUM);
+    moment_request.add_moments(CARTA::Moment::INTEGRATED_OF_THE_SPECTRUM);
+    moment_request.add_moments(CARTA::Moment::INTENSITY_WEIGHTED_COORD);
+    moment_request.add_moments(CARTA::Moment::INTENSITY_WEIGHTED_DISPERSION_OF_THE_COORD);
+    moment_request.add_moments(CARTA::Moment::MEDIAN_OF_THE_SPECTRUM);
+    moment_request.add_moments(CARTA::Moment::MEDIAN_COORDINATE);
+    moment_request.add_moments(CARTA::Moment::STD_ABOUT_THE_MEAN_OF_THE_SPECTRUM);
+    moment_request.add_moments(CARTA::Moment::RMS_OF_THE_SPECTRUM);
+    moment_request.add_moments(CARTA::Moment::ABS_MEAN_DEVIATION_OF_THE_SPECTRUM);
+    moment_request.add_moments(CARTA::Moment::MAX_OF_THE_SPECTRUM);
+    moment_request.add_moments(CARTA::Moment::COORD_OF_THE_MAX_OF_THE_SPECTRUM);
+    moment_request.add_moments(CARTA::Moment::MIN_OF_THE_SPECTRUM);
+    moment_request.add_moments(CARTA::Moment::COORD_OF_THE_MIN_OF_THE_SPECTRUM);
+    return moment_request;
+}
+
+CARTA::EventType Message::EventType(std::vector<char>& message) {
+    carta::EventHeader head = *reinterpret_cast<const carta::EventHeader*>(message.data());
+    return static_cast<CARTA::EventType>(head.type);
+}
+
+void Message::FillHistogram(CARTA::Histogram* histogram, const carta::BasicStats<float>& stats, const carta::Histogram& hist) {
+    FillHistogram(histogram, hist.GetNbins(), hist.GetBinWidth(), hist.GetBinCenter(), hist.GetHistogramBins(), stats.mean, stats.stdDev);
+}
+
+void Message::FillHistogram(CARTA::Histogram* histogram, int num_bins, double bin_width, double first_bin_center,
```

Comment:
If we're going to keep these function signatures (which I think is fine for now), these should *not* be methods on the `Message` class, since they are not like the functions which return messages. They should be functions outside of the class (as they were before), but still defined in the same files (`Message.h` and `Message.cc`).